src/dev/build/lib/fs.js

```diff
@@ -83,6 +84,13 @@ export async function copy(source, destination) {
 
   // do a stat call to make sure the source exists before creating the destination directory
   await statAsync(source);
+  // Delete `destination` since `fs.copyFile` with the copy-on-write reflink/COPYFILE_FICLONE
+  // flag fails if the destination already exists. https://github.com/nodejs/node/issues/27273
+  try {
+    await unlinkAsync(destination);
```

Comment:
Is this still necicary after cleaning the `.node_binaries` directory? My concern is the overhead of an additional delete call before any copy when it's not needed.

Reply:
cleaning `node_binaries` solves the original problem, so we don't need to add the unlink call.

I kept it there since it felt like a bug in our fs.copy() implementation, because we abstract away how we do the copy, I would expect this to behave like Node's fs.copy and then it weirdly doesn't without it being obvious from the calling code that we're passing in `COPYFILE_FICLONE`.

Since the whole point of using `COPYFILE_FICLONE` is for performance it makes sense that we want to prevent unnecessary calls.

I'm thinking perhaps it's best to rename the function to `copyOnWrite` and add a comment explaining the behaviour, but removing the unlink call.